compiler/record-minmax/src/RecordMinMax.cpp

```diff
@@ -43,12 +61,42 @@ void RecordMinMax::initialize(const std::string &input_model_path)
     throw std::runtime_error("ERROR: Failed to load '" + input_model_path + "'");
   }
 
-  // TODO: Initialize profiling-supported interpreter
+  // Initialize interpreter
+  _interpreter = std::make_unique<luci_interpreter::Interpreter>(_module.get());
+
+  // TODO: Attach observer to the interpreter
 }
 
 void RecordMinMax::profileData(const std::string &input_data_path)
 {
-  // TODO: Collect min/max data for the given input data (.h5)
+  HDF5Importer importer(input_data_path);
+
+  const auto num_records = importer.numRecords();
+  const auto input_nodes = loco::input_nodes(_module->graph());
+  const auto num_inputs = input_nodes.size();
+
+  for (int32_t i = 0; i < num_records; i++)
+  {
+    if (num_inputs != importer.numInputs(i))
+      throw std::runtime_error("Wrong number of inputs.");
+
+    for (int32_t j = 0; j < num_inputs; j++)
+    {
+      const auto *input_node = loco::must_cast<const luci::CircleInput *>(input_nodes[j]);
+      DataType dtype;
+      std::vector<char> input_data(getTensorSize(input_node));
+      importer.read(i, j, &dtype, input_data.data());
+
+      if (dtype != input_node->dtype())
```

Comment:
Please check shapes. This was the whole point of using hdf5.

Reply:
I am not sure about necessity of shape checks (that would be great, though), I suppose it depends on source of hdf5 files.

I propose to implement at least check that DataSet size is equal to size expected by model. Because size difference could crash this importer, if hdf5 DataSet is larger then buffers prepared on line 87.

P.S. Shape checks could be useful for variable input tensors. But still, it depends on data source and our agreement how to handle variable sizes.

Reply:
@binarman 
> I am not sure about necessity of shape checks (that would be great, though), I suppose it depends on source of hdf5 files.

See https://github.com/Samsung/ONE/issues/1605#issuecomment-637231643

Reply:
Added codes to check type and shape. Thanks!